tests/jenkins/TestOpenSearchReleasePromotionParallelTest.groovy

```diff
@@ -0,0 +1,168 @@
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_PLATFORM, value=linux})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_NAME, value=rpm})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_ARCHITECTURE, value=arm64})")
+
+        // OpenSearch Dashboards Linux rpm x64
+        assertCallStack().contains('stage(OpenSearch Dashboards Linux rpm x64, groovy.lang.Closure)')
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_JOB_NAME, value=distribution-build-opensearch-dashboards})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_PLATFORM, value=linux})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_NAME, value=rpm})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_ARCHITECTURE, value=x64})")
+
+        // OpenSearch Linux tar arm64
+        assertCallStack().contains('stage(OpenSearch Linux tar arm64, groovy.lang.Closure)')
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_JOB_NAME, value=distribution-build-opensearch})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_PLATFORM, value=linux})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_NAME, value=tar})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_ARCHITECTURE, value=arm64})")
+
+        // OpenSearch Dashboards Linux tar arm64
+        assertCallStack().contains('stage(OpenSearch Dashboards Linux tar arm64, groovy.lang.Closure)')
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_JOB_NAME, value=distribution-build-opensearch-dashboards})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_PLATFORM, value=linux})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_NAME, value=tar})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_ARCHITECTURE, value=arm64})")
+
+        // OpenSearch Dashboards Linux tar x64
+        assertCallStack().contains('stage(OpenSearch Dashboards Linux tar x64, groovy.lang.Closure)')
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_JOB_NAME, value=distribution-build-opensearch-dashboards})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_PLATFORM, value=linux})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_NAME, value=tar})")
+        assertCallStack().contains("release-promotion-parallel.string({name=DISTRIBUTION_ARCHITECTURE, value=x64})")
+
+        // OpenSearch Linux tar x64
```

Comment:
How do we know we have covered all the required steps in workflow like this? Is there a separate workflow model or state machine which can be source of truth?

Reply:
The `wait: true` parameter is responsible for returning the status of the triggered job. It propagates back the state.